### ISSUE TITLE
Deal with the "infrastructure files" in a nice way

### ISSUE THREAD

**Issue by necessarily-equal** [Unvanquished/Unvanquished#1423]
So, there are various files used to make a release build, and are needed to make distro packages, flatpak and stuff. And it's a bit of a mess. We need to unify

* the desktop file, which can be found in the updater repo
* the [appstream](https://github.com/Unvanquished/Unvanquished/pull/1422) soon to be in this repo. But @illwieckz suggested maybe it should be in another one?
* the icons, which are in this repo
* the binary crap, https://dl.unvanquished.net/deps/linux64-5.tar.bz2
* the daemon repo (illwieckz on IRC: « for the source repository, better use Dæmon repository. / only building the daemon engine is required. / that would also reduce the amount of deps to pull to make cmake happy »).
* the gamelogic and assets, that you can grab from https://dl.unvanquished.net/release/unvanquished_0.52.0.zip if you throw away the other part of the zip.

So that means to package a build, one has to grab things from 5 source (and will maybe need 6 sources if we split the appstream). That's not good.

* [ ] the desktop file(s) should be moved to the main repo, IMO
* [ ] (the appstream is still under discussion)
* [x] for the icons are good like this
* hopefully the NaCl binary crap will go away eventually, so let's ignore that.
* daemon is definitely split on purpose, maybe we'll want to change that someday, but let's say it's fine like that for now
* [ ] for the assets, maybe we can have a zip with just the dpk, not sure if we can do better?

**Comment by illwieckz** [comment]
I wonder if we can do a special repository for distro package stuff stuff.

Put the debian, arch, flatpak (and others if exist) thing in one place.

I don't mind if updater repo has his own copy (or variant) of the `.desktop` file and icons, and if Unvanquished repo has its own copy of the icon (for the Windows `Unvanquished.exe`).

What looks important to me is that all distros can get common things in one place.

**Comment by necessarily-equal** [comment]
That distro can be unvanquished I think, unless that "update the 
appimage" thing is too annoying. On the long term, if we want to 
translate the appimage and the desktop file, we'll probably be better 
off with something we can update?

**Comment by illwieckz** [comment]
Did you meant `s/distro/repo/;s/appimage/appstream/g`?

**Comment by necessarily-equal** [comment]
woops, yeah. Fixed. :sleeping:

**Comment by slipher** [comment]
> the desktop file, which can be found in the updater repo

The updater desktop file launches the updater, not the engine, so it's questionable to use it for a distro package (although it may work by accident by just replacing the binary name?). There is the protocol handler one which does launch the engine, but that's really a bug...

> the binary crap, https://dl.unvanquished.net/deps/linux64-5.tar.bz2

A Linux engine packager should not actually need the stuff from https://dl.unvanquished.net/deps/linux64-5.tar.bz2. I think you can prevent it from being downloaded by just creating the `external_deps/linux64-5/` directory before running CMake.

**Comment by slipher** [comment]
Right... you don't need the toolchain, but you need to copy over some of the binaries which make the runtime environment.
